server/src/main/java/org/opensearch/index/compositeindex/datacube/startree/builder/BaseStarTreeBuilder.java

```diff
@@ -379,26 +534,58 @@ private void createSortedDocValuesIndices(DocValuesConsumer docValuesConsumer, A
                 }
             }
         }
-
         addStarTreeDocValueFields(docValuesConsumer, dimensionWriters, dimensionFieldInfoList, numDimensions);
         addStarTreeDocValueFields(docValuesConsumer, metricWriters, metricFieldInfoList, metricAggregatorInfos.size());
     }
 
+    /**
+     * Adds startree field to respective field writers
+     */
+    private void indexDocValue(DocValuesWriterWrapper<?> dvWriter, int docId, long value, String field) throws IOException {
+        if (dvWriter instanceof SortedSetDocValuesWriterWrapper) {
+            // TODO : cache lookupOrd to make it faster
```

Comment:
would be a good optimization, but effects on JVM would be important to monitor for high cardinality workloads. Maybe something like frequently used cached elements of limited size?

Reply:
Yes that's the plan, to use a fixed size cache. Coming in a successive PR.